wallet-new/src/Cardano/Wallet/Kernel/Addresses.hs

```diff
@@ -49,10 +52,17 @@ data CreateAddressError =
     | CreateAddressHdRndAddressSpaceSaturated HdAccountId
       -- ^ The available number of HD addresses in use in such that trying
       -- to find another random index would be too expensive
-    deriving Eq
+    deriving (Generic, Eq)
+
+instance Aeson.ToJSON CreateAddressError
+instance Aeson.FromJSON CreateAddressError
 
 instance Arbitrary CreateAddressError where
-    arbitrary = oneof []
+    arbitrary = oneof [ CreateAddressUnknownHdAccount <$> arbitrary
```

Comment:
All of these `Arbitrary` instances could be replaced with `genericArbitrary`. Would rather move fast with this pr first though.

I think `genericArbitrary` would be really nice though, as a small amount of existing instances looked incomplete, as in "let's get this compiling", and not "this is exactly the arbitrary instance we need".